### PR TITLE
US CPI Data adapter

### PR DESCRIPTION
**Pull request by bogos**

### CODE REVIEW

File: packages/sources/uscpi-one/src/endpoint/series.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { Config, ExecuteWithConfig, InputParameters } from '@chainlink/types'
+
+export const supportedEndpoints = ['series']
+
+export const endpointResultPaths = {
+  example: 'series',
```
Comment (boxhock) [inline packages/sources/uscpi-one/src/endpoint/series.ts:7]: Here, the key should be the endpoint name (example -> series) and the value should be the path.

File: packages/sources/uscpi-one/src/endpoint/series.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  year: false,
+  month: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, inputParameters)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const serie = validator.validated.serie || 'CUSR0000SA0'
```
Comment (boxhock) [inline packages/sources/uscpi-one/src/endpoint/series.ts:46]:
```suggestion
  const serie = validator.validated.data.serie || 'CUSR0000SA0'
```

File: packages/sources/uscpi-one/src/endpoint/series.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  const month = validator.validated.data.month
+    ? capitalizeFirstLetter(validator.validated.data.month)
+    : ''
+  const url = `/timeseries/data/${serie}`
+  const options = { ...config.api, url }
+  const response = await Requester.request<ResponseSchema>(options, customError)
+  const data = response.data.Results.series[0].data
+
+  let filter
+  if (!year && !month) {
```
Comment (boxhock) [inline packages/sources/uscpi-one/src/endpoint/series.ts:57]: Can we document that only supplying year or month isn't going to do anything unless they're both included?

File: packages/sources/uscpi-one/src/endpoint/series.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    filter = data.filter((obj: DataSchema) => {
+      return obj['latest'] === 'true'
+    })
+  } else {
+    filter = data.filter((obj: DataSchema) => {
+      return obj['year'] === year && obj['periodName'] === month
+    })
+  }
+  const result = round(Requester.validateResultNumber(filter, [0, 'value']), 1)
```
Comment (boxhock) [inline packages/sources/uscpi-one/src/endpoint/series.ts:66]: We're not making the result path configurable here. Also, why are we rounding the result?

File: packages/sources/uscpi-one/test/integration/adapter.test.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  const execute = makeExecute()
+
+  describe('successful calls @integration', () => {
+    const requests = [
+      {
+        name: 'id not supplied',
+        testData: { data: { serie: 'EIUIR', month: 'july', year: '2021' } },
+      },
+      {
+        name: 'unknown serie',
```
Comment (boxhock) [inline packages/sources/uscpi-one/test/integration/adapter.test.ts:17]:
```suggestion
        name: 'serie not supplied',
```

File: packages/sources/uscpi-one/test/integration/adapter.test.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        testData: { id: jobID, data: { month: 'july', year: '2021' } },
+      },
+      {
+        name: 'unknown data',
```
Comment (boxhock) [inline packages/sources/uscpi-one/test/integration/adapter.test.ts:21]:
```suggestion
        name: 'empty data supplied',
```

File: packages/sources/uscpi-one/test/integration/adapter.test.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        testData: { id: jobID, data: {} },
+      },
+      {
+        name: 'unknown month and year',
```
Comment (boxhock) [inline packages/sources/uscpi-one/test/integration/adapter.test.ts:25]:
```suggestion
        name: 'month and year not supplied',
```

File: packages/sources/uscpi-one/src/config.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { Requester } from '@chainlink/ea-bootstrap'
+import { Config } from '@chainlink/types'
+
+export const NAME = 'USCPIONE'
```
Comment (boxhock) [inline packages/sources/uscpi-one/src/config.ts:4]: The adapter path is `uscpi-one`, which to me makes sense that the name would be `USCPI_ONE`.

**Review by boxhock — CHANGES_REQUESTED** [review]

File: packages/sources/uscpi-one/test/integration/adapter.test.ts (hunk 7 above)
Comment (bogos) [reply packages/sources/uscpi-one/test/integration/adapter.test.ts]: done

File: packages/sources/uscpi-one/src/config.ts (hunk 8 above)
Comment (bogos) [reply packages/sources/uscpi-one/src/config.ts]: done

File: packages/sources/uscpi-one/src/endpoint/series.ts (hunk 4 above)
Comment (bogos) [reply packages/sources/uscpi-one/src/endpoint/series.ts]: added the result path, removed the round result

File: packages/sources/uscpi-one/src/endpoint/series.ts (hunk 1 above)
Comment (bogos) [reply packages/sources/uscpi-one/src/endpoint/series.ts]: done

File: packages/sources/uscpi-one/src/endpoint/series.ts (hunk 3 above)
Comment (bogos) [reply packages/sources/uscpi-one/src/endpoint/series.ts:60]: done :)

File: packages/sources/uscpi-one/README.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+### Input Parameters
+
+| Required? |   Name   |     Description     |         Options          | Defaults to |
+| :-------: | :------: | :-----------------: | :----------------------: | :---------: |
+|           | endpoint | The endpoint to use | [serie](#Serie-Endpoint) |    serie    |
+
+---
+
+## Serie Endpoint
```
Comment (boxhock) [inline packages/sources/uscpi-one/README.md:13]: The README says the endpoint name is "serie" but the code uses endpoint name "series"

File: packages/sources/uscpi-one/README.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
+# Chainlink External Adapter for US Consumer Price Index (USCPI)
+
+---
+
+### Input Parameters
+
+| Required? |   Name   |     Description     |         Options          | Defaults to |
+| :-------: | :------: | :-----------------: | :----------------------: | :---------: |
+|           | endpoint | The endpoint to use | [serie](#Serie-Endpoint) |    serie    |
```
Comment (boxhock) [inline packages/sources/uscpi-one/README.md:9]: Same here, code uses "series"

**Review by boxhock — CHANGES_REQUESTED** [review]
Just a small README change needed, otherwise LGTM

File: yarn.lock [hunk 11]
```diff
@@ -1685,6 +1685,21 @@ __metadata:
...
+    "@chainlink/types": 0.0.1
+    "@types/jest": ^27.0.1
+    "@types/node": ^14.14.35
+    axios: ^0.21.1
+    sinon: ^11.1.2
+    tslib: ^2.3.1
+    typescript: ^4.3.5
+  languageName: unknown
+  linkType: soft
+
```
Comment (boxhock) [inline yarn.lock:1702]: This should probably not be included in this PR. This is breaking CI

**Review by boxhock — CHANGES_REQUESTED** [review]

**Pull request closed without merging**